src/main/perf/LineFileDocs.java

```diff
@@ -165,9 +167,10 @@ private void readDocs() throws Exception {
           throw new RuntimeException("expected " + length + " document bytes but read " + x);
         }
         buffer.position(0);
-        queue.put(new LineFileDoc(buffer, readVector(count)));
+        queue.put(new LineFileDoc(buffer, readVector(docCountPerBlock), idBase));
```

Comment:
Hmm if this (`idBase`) is the base, and not incrementing with each doc, aren't we passing the same `id` to `readVector` multiple times in a row?

Reply:
This base is actually for `doRepeat` case, basically if we repeat the document then we need a `base == num_epoch * max_doc` to distinguish doc from last epoch and doc from current epoch. So in most cases it will just be 0.